partner_address_street3/hooks.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014 Nicolas Bessi, Alexandre Fayolle, Camptocamp SA
+# Copyright 2016 Sodexis
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+def post_init_hook(cr, registry):
+    query = """
+        UPDATE res_country
+        SET address_format = replace(
+        address_format,
+        E'%(street2)s\n',
+        E'%(street2)s\n%(street3)s\n'
+        )
+    """
+    cr.execute(query)
+
+
+def uninstall_hook(cr, registry):
+    query = """
+        UPDATE res_country
+        SET address_format = replace(
+        address_format,
+        E'%(street3)s\n',
```

Comment:
if I make it drop %(street3) pattern only, then address format will have an extra line after the module is uninstalled, to resolve this I need to have 2 replace query for patterns E'%(street3)s\n' and E'%(street3)s'.

Let me know if you have any solution to this.